R/get_dupes.R

```diff
@@ -3,47 +3,48 @@
 #' @description
 #' For hunting duplicate records during data cleaning.  Specify the data.frame and the variable combination to search for duplicates and get back the duplicated rows.
 #'
-#' @param dat the input data.frame.
-#' @param ... unquoted variable names to search for duplicates.
+#' @param dat The input data.frame.
+#' @param ... Unquoted variable names to search for duplicates. This takes a tidyselect specification.
 #' @return Returns a data.frame (actually a \code{tbl_df}) with the full records where the specified variables have duplicated values, as well as a variable \code{dupe_count} showing the number of rows sharing that combination of duplicated values.
 #' @export
 #' @examples
 #' get_dupes(mtcars, mpg, hp)
 #' # or called with the magrittr pipe %>% :
 #' mtcars %>% get_dupes(wt)
+#' # or using tidyselect:
+#' mtcars %>% get_dupes(weight = wt, starts_with("cy"))
 #'
 
 get_dupes <- function(dat, ...) {
-  uq_names <- as.list(substitute(list(...)))[-1L] # unquoted names for NSE calls, need quoted names separately for messages + warnings
-  df_name <- deparse(substitute(dat))
-
-  if (length(uq_names) == 0) { # if called on an entire data.frame with no specified variable names
+  
+  expr <- rlang::expr(c(...))
+  pos <- tidyselect::eval_select(expr, data = dat)
+  
+  names(dat)[pos] <- names(pos) #allows for renaming within get_dupes() consistent with select()
+  
+  if (rlang::dots_n(...) == 0) { # if no tidyselect variables are specified, check the whole data.frame
     var_names <- names(dat)
     nms <- rlang::syms(var_names)
     message("No variable names specified - using all columns.\n")
   } else {
-    nms <- rlang::quos(...) %>%
-      rlang::quos_auto_name()
-    var_names <- names(nms)
+    var_names <- names(pos)
+    nms <- rlang::syms(var_names)
   }
-
-  # check that each variable name provided is present in names(dat); if not, throw error
-  check_vars_in_df(dat, df_name, var_names)
```

Comment:
is this removed b/c it was used to deal with how the old tidyverse functions errored and now they error in the desired way already -- i.e., they identify the mis-specified variable -- so it's now redundant?

Reply:
Correct.  Tidyselect will throw its own error if an exact column specification doesn't match the table.

But just a reminder (we'e discussed previously) that any matching functions (starts_with, etc) that don't catch any variables will not generate any error currently.